internal/generate/clusterserviceversion/bases/definitions/ast.go

```diff
@@ -66,9 +66,11 @@ func (g generator) getMarkedChildrenOfField(root markers.FieldInfo) (map[string]
 						}
 						// Create a new set of path segments using the parent's segments
 						// and add the field to the next fields to search.
+						parentSegments := make([]string, len(field.pathSegments))
```

Comment:
`append` won't create a new underlying array if its cap is sufficient
```suggestion
						parentSegments := make([]string, len(field.pathSegments), len(field.pathSegments)+1)
```